lib/internal/streams/duplexify.js

```diff
@@ -83,15 +83,19 @@ module.exports = function duplexify(body, name) {
   }
 
   if (typeof body === 'function') {
-    const { value, write, final, destroy } = fromAsyncGen(body);
+    let d;
+
+    const { value, write, final, destroy } = fromAsyncGen(body, () => {
+      if (d) destroyer(d);
```

Comment:
```suggestion
     destroyer(d);
```
The destorer function already checks whether the stream exists:
https://github.com/nodejs/node/blob/3c5ceff85fb90b265a85270bba38085f45c7f25e/lib/internal/streams/destroy.js#L323